src/main/java/org/gridsuite/actions/server/ContingencyListService.java

```diff
@@ -345,6 +371,16 @@ public Optional<FormContingencyList> createFormContingencyList(UUID sourceListId
         return Optional.empty();
     }
 
+    public Optional<IdBasedContingencyList> createIdBasedContingencyList(UUID sourceListId, UUID id) {
+        IdBasedContingencyList sourceIdBasedContingencyList = getIdBasedContingencyList(sourceListId).orElse(null);
+        if (sourceIdBasedContingencyList != null) {
```

Comment:
Why we do not use `createIdBasedContingencyList` method ?

Reply:
done for the 3 duplication methods